### ISSUE TITLE
Beta channel installations for arm64 fail - "Unable to determine Flutter version for channel: beta version: 3.19.0-0.3.pre architecture: arm64"

### ISSUE THREAD

**Issue by caffeineflo** [subosito/flutter-action#279]
Hi,

probably a very minor parsing bug, but I thought I'll file a ticket so this issue is known.

Here are the details:

Workflow parameters:
```
  with:
    channel: beta
    flutter-version: 3.19.0-0.3.pre
```

Runner: `macos-latest-xlarge` (M1, arm64 machine)

When running the installation step, the step fails reporting 

```
Run $GITHUB_ACTION_PATH/setup.sh -p -c '/Users/runner/hostedtoolcache/flutter/:channel:-:version:-:arch:' -k 'flutter-:os:-:channel:-:version:-:arch:-:hash:' -d 'flutter-pub:os:-:channel:-:version:-:arch:-:hash:' -n '3.19.0-0.3.pre' -a 'ARM64' beta
  $GITHUB_ACTION_PATH/setup.sh -p -c '/Users/runner/hostedtoolcache/flutter/:channel:-:version:-:arch:' -k 'flutter-:os:-:channel:-:version:-:arch:-:hash:' -d 'flutter-pub:os:-:channel:-:version:-:arch:-:hash:' -n '3.19.0-0.3.pre' -a 'ARM64' beta
  shell: /bin/bash --noprofile --norc -e -o pipefail {0}
##[debug]/bin/bash --noprofile --norc -e -o pipefail /Users/runner/work/_temp/d7c7c8dc-0a99-46e0-a311-6ac7b1a8a8a7.sh
Unable to determine Flutter version for channel: beta version: 3.19.0-0.3.pre architecture: arm64
```

I checked the version on the flutter sdk archive, which exists and the string is correct. I assume there's a parsing issue somewhere that makes this fail. 

Let me know if any other information is needed.

**Comment by koji-1009** [comment]
@caffeineflo 
Hi, I checked the released version and it seems that 3.19.0-0.3.pre + arm64 does not exist.
https://docs.flutter.dev/release/archive?tab=macos

Could you test with `3.19.0-0.4.pre`?

**Comment by caffeineflo** [comment]
@koji-1009 You are right! I could have sworn it was there when I checked last week, but this makes sense then! Thanks
